### ISSUE TITLE
Voorbeeld url om de api te bevragen

### ISSUE THREAD

**Issue by smartman63** [VNG-Realisatie/Haal-Centraal-BRK-bevragen#717]
Heeft er iemand een voorbeeld om de BRK Api te bevragen. Ik doe dit met behulp van Postman maar krijg het niet voor elkaar.
Ik begin hier (in Postman): {{baseUrl}}/kadastraalonroerendezaken?postcode=<string>
Ik heb de volgende url om de testomgeving te benaderen: https://api.brk.kadaster.nl/esd-eto-apikey/bevragen/v1/
Wordt dit dan de "baseUrl" en waar vul ik mijn verkregen API key in. Moet dat in de url of moet dat in het Header deel in Postman worden ingevuld?

**Comment by fsamwel** [comment]
De variabele {{baseUrl}} moet de waarde https://api.brk.kadaster.nl/esd-eto-apikey/bevragen/v1 krijgen, dus zonder de / aan het einde.

Daarnaast moet de API Key in de requestheader "X-API-KEY" worden meegestuurd.

Bijvoorbeeld:
curl --location --request GET 'https://api.brk.kadaster.nl/esd-eto-apikey/bevragen/v1/kadastraalonroerendezaken/76870488070000' \
--header 'X-API-KEY: 1k2j35hk1j23h45k1j23h4kj123h4kj12hk4jh12k34'

**Comment by smartman63** [comment]
Dankjewel @fsamwel . 
![Naamloos](https://user-images.githubusercontent.com/10928430/96582594-cb20e300-12db-11eb-9df2-99287ded8fbf.jpg)
Ik heb de baseUrl aangepast zoals je beschrijft en de X-API-KEY geconfigureerd in het Header-deel in Postman. Is dat dan de juiste manier want ik krijg geen connectie lijkt het. Er schijnt ook iets met de permissies te zijn. zie mijn bijgesloten afbeelding, rechtsonder; de melding van de Agent. Ik hoor het graag.

**Comment by fsamwel** [comment]
Beste @smartman63 bij de headers kan je ook autonmatisch gegenereerde headers zien. Althans bij mij toont Postman ook een knopje "8 Hidden". Wanneer ik die klik krijg ik ook de headers van deze screenprint te zien. Heb jij die ook?

![postman headers](https://user-images.githubusercontent.com/37143495/96591344-74210b00-12e7-11eb-89c5-30556841e9ac.png)

**Comment by smartman63** [comment]
Hoi @fsamwel. Nou in eerste instantie niet maar na de installatie van de Postman Agent wel. Ik heb een screenshot bijgesloten.
![Naamloos_1](https://user-images.githubusercontent.com/10928430/96680922-b04f7c80-1376-11eb-9116-71b140715d77.png)
Ik zie echter in mijn screenshot niet echt een "KEY-verschil" tussen jouw en mijn gegevens. Kan je wellicht mij eens een aanroep sturen die bij jou werkt. Uiteraard zonder jouw key. Op die plek vul ik dan mijn eigen key in. Als dat dan niet werkt dan moet het ergens in de autorisatie zitten zoals mijn screenshot in de Agent ook aangeeft. Dan is wellicht mijn Api-key niet werkend of iets dergelijks.

**Comment by MelvLee** [comment]
Gezien de foutmelding uit de eerste screenshot, denk ik dat je last heb van [CORS (Cross-Origin Resource Sharing)](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS).

In het kort komt het er op neer dat je browser een check uitvoert bij api.brk.kadaster.nl of een request vanaf martkevin.postman.co domein mag en gezien de foutmelding mag dat niet.

Ik denk dat je voor nu een paar dingen kan proberen om te zien of het aan jouw api key ligt
- de **host** checkbox disablen en kijken of het dan werkt. Zo niet, dan
- via getpostman.com de postman applicatie downloaden en het daarmee proberen

**Comment by smartman63** [comment]
Hoi. Daar ben ik weer. Op aanraden van @MelvLee heb ik de host uitgezet: werkt wederom niet. Hierna heb ik de applicatie lokaal geïnstalleerd en ook hier krijg ik het niet aan de gang.

![Naamloos2](https://user-images.githubusercontent.com/10928430/96697784-99b42000-138c-11eb-987f-58f79b775320.jpg)


Ik krijg als de host uit staat geen melding over autorisatie, nu lijkt het echter dat de aanroep verkeerd is. Bad request. 
Nog ideeën? Het duizelt me nu een beetje :-)

**Comment by fsamwel** [comment]
volgens mij doe je niks anders dan ik. De url klopt, je gebruikt de juiste x-api-key header.

Krijg je nog steeds de CORS melding (dus vanuit de Postman applicatie)? Het klopt dat de API geen requests toestaat vanuit een browser, maar vanuit de applicatie zou je die denk ik niet moeten krijgen.

**Comment by smartman63** [comment]
Nope. Geen CORS melding. Die had ik alleen in de webbrowser omgeving. Nu alleen maar http 400 Bad request.

**Comment by fsamwel** [comment]
dan ben je een stap verder, de 401 ben je voorbij. Wat is de response body bij de 400 bad request?

**Comment by smartman63** [comment]
Waar zie ik dat? Hier...(rode gebied), zie afbeelding. Is leeg of is die "1" een Response?
![Naamloos_2](https://user-images.githubusercontent.com/10928430/96716535-70ec5480-13a5-11eb-9e84-4ea7ac70872b.png)

**Comment by fsamwel** [comment]
Je hebt hier inderdaad een lege response. Heel vreemd, kan ik niet reproduceren.

Zou je een screenshot van wat in Params staat en van wat in de Settings staat kunnen toevoegen? Misschien dat ik daar verschillen zie.

Hoe heb je eigenlijk dit Postman project gemaakt? Vanuit de [OAS](https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-BRK-bevragen/master/specificatie/genereervariant/openapi.yaml)? Of door importeren van de [Postman collection](https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-BRK-bevragen/master/test/BRK-Bevragen-postman-collection.json)?
